src/server/game/Spells/Spell.cpp

```diff
@@ -4739,6 +4739,8 @@ SpellCastResult Spell::CheckCast(bool strict)
         {
             if (m_triggeredByAuraSpell)
                 return SPELL_FAILED_DONT_REPORT;
+            else if (m_spellInfo->HasAttribute(SPELL_ATTR0_DISABLED_WHILE_ACTIVE))  // Shadowform, Tricks of the Trade, Misdirection
```

Comment:
Failing SpellHistory::IsReady check should continue to return SPELL_FAILED_NOT_READY - it means the cast failed because it was already active or we deactivated it and triggered normal cooldown. Sending any other value than SPELL_FAILED_NOT_READY will make the client clear cooldown on it.